### ISSUE TITLE
`override` becomes `member` in interface implementations

### ISSUE THREAD

**Issue by auduchinok** [fsprojects/fantomas#263]
```fsharp
open System

type T() =
    interface IDisposable with
        override x.Dispose() = ()
```
becomes
```fsharp
module Scratch_6

open System

type T() =
    interface IDisposable with
        member x.Dispose() = ()
```

**Comment by nojaf** [comment]
It appear both `override` and `member` produce the exact same AST:

```fsharp
        Interface
                        (LongIdent (LongIdentWithDots ([IDisposable],[])),
                         Some
                           [Member
                              (Binding
                                 (None,NormalBinding,false,false,[],
                                  PreXmlDoc
                                    ((6,32),
                                     Microsoft.FSharp.Compiler.Ast+XmlDocCollector),
                                  SynValData
                                    (Some {IsInstance = true;
                                           IsDispatchSlot = false;
                                           IsOverrideOrExplicitImpl = true;
                                           IsFinal = false;
                                           MemberKind = Member;},
                                     SynValInfo
                                       ([[SynArgInfo ([],false,None)]; []],
                                        SynArgInfo ([],false,None)),None),
                                  LongIdent
                                    (LongIdentWithDots
                                       ([x; Dispose],
```

I'm guessing there is no real difference so we should perhaps choose whatever [the style guide](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/) says.
@cartermp could you provide some input here?

**Comment by nojaf** [comment]
Interesting remark @jindraivanek , I did not know it is possible to know what text fits what AST.
Could you elaborate the following code at bit:
```fsharp
| ElIf((e1,e2, _, _)::es, enOpt) ->
        atCurrentColumn (!- "if " +> ifElse (checkBreakForExpr e1) (genExpr astContext e1 ++ "then") (genExpr astContext e1 +- "then") -- " " 
            +> preserveBreakNln astContext e2
            +> fun ctx -> col sepNone es (fun (e1, e2, r, fullRange) ->
                             let elsePart =
                                 ifElse (ctx.Comments.ContainsKey fullRange.Start)
                                    (!+ "else" +> indent +> sepNln -- "if ")
                                    (!+ "else if ")                                        
                             ifElse (startWith "elif" r ctx) (!+ "elif ") elsePart
                             +> ifElse (checkBreakForExpr e1) (genExpr astContext e1 ++ "then") (genExpr astContext e1 +- "then") 
                             -- " " +> preserveBreakNln astContext e2) ctx
            +> opt sepNone enOpt (fun en -> !+ "else " +> preserveBreakNln astContext en))
```

Is `ElIf` a `SynExpr`? `r` is a [range](https://fsharp.github.io/FSharp.Compiler.Service/reference/microsoft-fsharp-compiler-range-range.html) in this case right?

`startWith` will check the range of the AST and `lookup` the text from `Context.Config` to see if it indeeds starts with the given prefix (`"elif"` in this case).

Is my reasoning correct here?

**Comment by jindraivanek** [comment]
@nojaf Yes, this is also my understanding.

`ElIf` is (recursive) active pattern https://github.com/fsprojects/fantomas/blob/master/src/Fantomas/SourceParser.fs#L788, that simplify nested `SynExpr.IfThenElse` into list. `e1`, `e2` are `SynExpr`.

`r` is indeed range.

In this snippet you can also see, that range can be also used to check if there is comment before range start (`ctx.Comments.ContainsKey fullRange.Start`). This is my change, see #241 :)

`lookup` is also used in other places - consts are copied from original code, unless `strict-mode` is used: https://github.com/fsprojects/fantomas/blob/master/src/Fantomas/SourceTransformer.fs#L126

**Comment by nojaf** [comment]
I see, thanks for explaining. 
Early guess is that the fix should be somewhere in [MemberBinding](https://github.com/fsprojects/fantomas/blob/master/src/Fantomas/CodePrinter.fs#L396)

**Comment by jindraivanek** [comment]
Fixed by #290.
